tests/shared/core/test_events.py

```diff
@@ -614,3 +621,160 @@ def test_event_executed_comparison(
+            text="hello",
+            parse_data={
+                "intent": {"id": 2, "name": "greet", "confidence": 0.9604260921478271,},
+                "entities": [
+                    {"entity": "city", "value": "London"},
+                    {"entity": "count", "value": 1},
+                ],
+                "text": "hi",
+                "message_id": "3f4c04602a4947098c574b107d3ccc50",
+                "metadata": {},
+                "intent_ranking": [
+                    {"id": 2, "name": "greet", "confidence": 0.9604260921478271,},
+                    {"id": 1, "name": "goodbye", "confidence": 0.01835782080888748,},
+                    {"id": 0, "name": "deny", "confidence": 0.011255578137934208,},
+                    {
+                        "id": 3,
+                        "name": "bot_challenge",
+                        "confidence": 0.004019865766167641,
+                    },
+                    {"id": 4, "name": "affirm", "confidence": 0.002524246694520116,},
+                    {
+                        "id": 5,
+                        "name": "mood_great",
+                        "confidence": 0.002214624546468258,
+                    },
+                    {"id": 6, "name": "chitchat", "confidence": 0.0009614597074687481,},
+                    {
+                        "id": 7,
+                        "name": "mood_unhappy",
+                        "confidence": 0.00024030178610701114,
+                    },
+                ],
+                "response_selector": {
+                    "all_retrieval_intents": [],
+                    "default": {
+                        "response": {
+                            "id": 11,
+                            "responses": [{"text": "chitchat/ask_name"}],
+                            "response_templates": [{"text": "chitchat/ask_name"}],
+                            "confidence": 0.9618658423423767,
+                            "intent_response_key": "chitchat/ask_name",
+                            "utter_action": "utter_chitchat/ask_name",
+                            "template_name": "utter_chitchat/ask_name",
+                        },
+                        "ranking": [
+                            {
+                                "id": 11,
+                                "confidence": 0.9618658423423767,
+                                "intent_response_key": "chitchat/ask_name",
+                            },
+                            {
+                                "id": 12,
+                                "confidence": 0.03813415765762329,
+                                "intent_response_key": "chitchat/ask_weather",
+                            },
+                        ],
+                    },
+                },
+            },
+        ),
+        SessionStarted(),
+        ActionExecuted(action_name="action_listen"),
+        AgentUttered(),
+    ],
+)
+def test_event_fingerprint_consistency_across_runs(event: Event):
```

Comment:
Here I wasn't sure if to use `rasa.shared.utils.common.all_subclassess` to iterate through all the subclassess of Event in case we add Event types in the future? This would imply swapping the long list in `pytest.mark.parametrize` with a long if-elif because some events have some specific arguments that need to be passed.

Reply:
Yup, can do - however a warning that not all subclassess exist in this list list, for example internal events such as `WronglyPredictedAction`. I can filter these out in the test.